### ISSUE TITLE
Active learning loop: iterative data improvement

### ISSUE THREAD

**Issue by NP-compete** [NP-compete/smithery#21]
## Problem

The first round of training data won't be perfect. Some tools will have lower accuracy than others. Some argument patterns will be harder. Instead of manually identifying weaknesses and generating more data, smithery should automate this with an active learning loop.

### The Loop

\`\`\`
┌─────────────┐     ┌─────────┐     ┌──────────┐     ┌─────────────┐
│ Generate     │────▶│ Train   │────▶│ Evaluate │────▶│ Analyze     │
│ Data (N)     │     │         │     │          │     │ Failures    │
└─────────────┘     └─────────┘     └──────────┘     └──────┬──────┘
       ▲                                                      │
       │                                                      ▼
       │                                              ┌──────────────┐
       └──────────────────────────────────────────────│ Generate     │
                                                      │ Targeted Data│
                                                      └──────────────┘
\`\`\`

1. **Generate** initial training data (e.g., 5,000 examples)
2. **Train** model
3. **Evaluate** on test set
4. **Analyze failures**: which tools fail most? which argument patterns? which difficulty levels?
5. **Generate targeted data**: produce more examples specifically for the weak spots
6. **Repeat** until accuracy targets are met or improvement plateaus

### Failure Analysis

Automatic categorization of failures:
- **Wrong tool selected**: confusion matrix → generate disambiguating examples
- **Missing required param**: → generate more examples with that param pattern
- **Wrong param value**: → generate examples with diverse values for that param
- **Format error**: → generate examples with clean formatting emphasis
- **False refusal**: → generate more borderline-but-valid examples

### CLI

\`\`\`bash
smithery forge --mcp ./server.json --active-learning --target-accuracy 0.95 --max-rounds 5
\`\`\`

### Configuration

\`\`\`yaml
active_learning:
  enabled: true
  max_rounds: 5
  examples_per_round: 1000
  target_metrics:
    tool_accuracy: 0.95
    param_f1: 0.90
    safety_refusal: 0.99
  stop_when: all_targets_met  # or: no_improvement
\`\`\`

## Acceptance Criteria

- [ ] Failure analysis module categorizes evaluation errors
- [ ] Targeted data generation focuses on identified weaknesses
- [ ] Active learning loop runs automatically (generate → train → eval → repeat)
- [ ] Configurable stopping criteria (accuracy target or plateau detection)
- [ ] Per-round metrics logged and displayed
- [ ] CLI flag \`--active-learning\` enables the loop
- [ ] Report showing improvement curve across rounds
- [ ] Unit tests for failure analysis

## Dependencies

- Depends on #5 (data generation), #8 (training), #9 (evaluation)
- Natural extension of the forge pipeline

## References

- [Active learning survey](https://arxiv.org/abs/2009.09205) — Strategies for efficient labeling
- Curriculum learning literature — easy → hard example ordering

## Estimated Effort

Large (1-2 weeks). The failure analysis and targeted generation require significant experimentation to get right.
